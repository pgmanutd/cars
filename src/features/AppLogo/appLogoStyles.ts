import { makeStyles, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(() =>
  createStyles({
    root: {
      width: 64,
      height: 64,
    },
  }),
);
