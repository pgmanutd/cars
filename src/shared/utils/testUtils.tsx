import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
