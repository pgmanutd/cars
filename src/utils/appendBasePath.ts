const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
