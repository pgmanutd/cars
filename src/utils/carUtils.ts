import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
