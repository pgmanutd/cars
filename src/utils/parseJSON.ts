const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
