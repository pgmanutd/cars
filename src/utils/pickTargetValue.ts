import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
