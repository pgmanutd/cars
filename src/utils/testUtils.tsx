import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

import { TranslateProvider } from 'features/Translate';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  renderResult: render(
    <TranslateProvider>
      <HelmetProvider>
        <Router history={history}>{ui}</Router>
      </HelmetProvider>
    </TranslateProvider>,
  ),
  history,
});

export const renderWithTranslate = (ui: React.ReactElement) => ({
  renderResult: render(<TranslateProvider>{ui}</TranslateProvider>),
});
