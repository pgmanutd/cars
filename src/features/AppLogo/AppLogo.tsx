import React from 'react';
import CardMedia from '@material-ui/core/CardMedia';

import Logo from 'assets/images/logo.png';

import { useStyles } from './appLogoStyles';

const AppLogo: React.FC = (props) => {
  const classes = useStyles();

  return (
    <CardMedia
      data-testid="AppLogo"
      image={Logo}
      title="AppLogo"
      {...props}
      className={classes.root}
    />
  );
};

export default AppLogo;
