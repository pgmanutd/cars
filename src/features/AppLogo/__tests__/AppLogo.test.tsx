import React from 'react';

import { renderWithTranslate } from 'utils/testUtils';

import AppLogo from '../AppLogo';

describe('<AppLogo />', () => {
  const setup = () => {
    const { renderResult } = renderWithTranslate(<AppLogo />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
