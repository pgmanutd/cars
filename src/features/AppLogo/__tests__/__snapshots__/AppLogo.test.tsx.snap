// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLogo /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiCardMedia-root makeStyles-root-1"
    data-testid="AppLogo"
    style="background-image: url(logo.png);"
    title="AppLogo"
  />
</DocumentFragment>
`;
